File: 06-1-checkweather_fl/lib/service/weather_response.dart
class WeatherResponse {
  final double temp;
  final double feelsLike;
  final double tempMin;
  final double tempMax;
  final int pressure;
  final int humidity;
  final int seaLevel;

  WeatherResponse({
    required this.temp,
    required this.feelsLike,
    required this.tempMin,
    required this.tempMax,
    required this.pressure,
    required this.humidity,
    required this.seaLevel,
  });

  factory WeatherResponse.fromJson(Map<String, dynamic> json) {
    final main = json['main'];
    return WeatherResponse(
      temp: main['temp'].toDouble(),
      feelsLike: main['feels_like'].toDouble(),
      tempMin: main['temp_min'].toDouble(),
      tempMax: main['temp_max'].toDouble(),
      pressure: main['pressure'].toInt(),
      humidity: main['humidity'].toInt(),
      seaLevel: main['sea_level'].toInt(),
    );
  }

  @override
  String toString() {
    return 'WeatherResponse(temp: $temp, feelsLike: $feelsLike, tempMin: $tempMin, tempMax: $tempMax, pressure: $pressure, humidity: $humidity, seaLevel: $seaLevel)';
  }
}

File: kotlin_projects/06-1-CheckWeatherApp_kt2/app/src/main/java/com/example/retrofit1app_kt2/service/WeatherResponse.kt
package com.example.retrofit1app_kt2.service

import com.google.gson.annotations.SerializedName

data class WeatherResponse (
    val main: Main
)

data class Main(
    @SerializedName("temp")
    val temp: Double,

    @SerializedName("feels_like")
    val feelsLike: Double,

    @SerializedName("temp_min")
    val tempMin: Double,

    @SerializedName("temp_max")
    val tempMax: Double,

    @SerializedName("pressure")
    val pressure: Int,

    @SerializedName("humidity")
    val humidity: Int,

    @SerializedName("sea_level")
    val seaLevel: Int
)
